### DISCUSSION TITLE
How to disconnect all connected clients ?

### DISCUSSION THREAD

**Discussion by Axnjr** [Totodore/socketioxide#314] [Q&A] [answered]
I'm working on a project where in I want to disconnect all connected clients at once by emitting a `BEAT` event in some interval, the clients that are active would again connect to the server and inactive clients would stay disconnected. 

### Is there a built-in method or recommended pattern for achieving this?

**Comment by Totodore** [comment, marked as answer]
What do you mean by "active" and "inactive"? If it is simply clients that are not reachable due to a network bug there is already a mechanism for that and the socket will be disconnected. 
If it is a "state" change then you should simply emit an event to notify the server rather than emiting "BEAT" events. Once you receive this particular state you can add the socket to the room of your choice:
https://docs.rs/socketioxide/latest/socketioxide/handler/extract/struct.SocketRef.html#method.join

Later you can apply any operation on the sockets in the given room, for example disconnect the socket, or make them join another room, or remove them from a room:
https://docs.rs/socketioxide/latest/socketioxide/operators/struct.BroadcastOperators.html#method.disconnect

**Reply by Axnjr** [reply]
Sorry for confusion, I am new with WebSocket's din't knew what happens with unreachable clients

I want to keep track of concurrent connections. 

Suppose a user has a `max_connection_limit = 500`, and he reaches the limit then I wont allow new connections and I will revalidate the connection count in every hour using a lambda function.
